src/EditorFeatures/Core/Implementation/EditAndContinue/ActiveStatementTrackingService.cs

```diff
@@ -232,19 +262,25 @@ private async Task TrackActiveSpansAsync()
                         // The underlying text buffer has changed - this means that our tracking spans 
                         // are no longer useful, we need to refresh them. Refresh happens asynchronously 
                         // as we calculate document delta.
-                        SetTrackingSpansNoLock(document.Id, null);
+                        SetTrackingSpansNoLock(baseDocument.Id, null);
 
-                        // fire and forget:
-                        _ = RefreshTrackingSpansAsync(document, snapshot);
+                        // fire and forget on a background thread:
+                        try
+                        {
+                            _ = Task.Run(() => RefreshTrackingSpansAsync(baseDocument, document, snapshot), _editSession.CancellationToken);
+                        }
+                        catch (TaskCanceledException)
```

Comment:
Ditto: what's being thrown here? This will return a task that might be cancelled, since you're not observing it it doesn't matter.